### PR TITLE
Use streamly core 0.2

### PR DESCRIPTION
**Pull request by harendra-kumar**

### CODE REVIEW

File: src/Streamly/Statistics.hs [hunk 1]
```diff
@@ -215,6 +213,25 @@ import Prelude hiding (length, sum, minimum, maximum)
...
+length = Window.windowLength
+
+sum :: (Monad m, Num a) => Fold m (a, Maybe a) a
+sum = Window.windowSum
+
+sumInt :: (Monad m, Integral a) => Fold m (a, Maybe a) a
+sumInt = Window.windowSumInt
+
+powerSum :: (Monad m, Num a) => Int -> Fold m (a, Maybe a) a
+powerSum = Window.windowPowerSum
```
Comment (adithyaov) [inline src/Streamly/Statistics.hs:233]: Why did you redefine them?

File: streamly-statistics.cabal [hunk 2]
```diff
@@ -109,7 +109,7 @@ library
     import: ghc-options
     exposed-modules:     Streamly.Statistics
     build-depends:       base     >= 4.9 && < 5
-                       , streamly-core == 0.1.0
+                       , streamly-core >= 0.2.0
```
Comment (adithyaov) [inline streamly-statistics.cabal:112]: You can put a `< 0.3.0` upper bound.

File: src/Streamly/Statistics.hs [hunk 3]
```diff
@@ -215,6 +213,25 @@ import Prelude hiding (length, sum, minimum, maximum)
 -- TODO We have many of these functions in Streamly.Data.Fold as well. Need to
 -- think about deduplication.
 
+-------------------------------------------------------------------------------
+-- Re-exports
+-------------------------------------------------------------------------------
+
+lmap :: (c -> a) -> Fold m (a, Maybe a) b -> Fold m (c, Maybe c) b
```
[the remark this answers is not in the document]
Comment (harendra-kumar) [reply src/Streamly/Statistics.hs:220]: 1. that would be a breaking change
2. all combinators in this module are window, therefore, all of them will have to be prefixed

So this should be a separate change, if at all we want make that change.

File: streamly-statistics.cabal (hunk 2 above)
Comment (harendra-kumar) [reply streamly-statistics.cabal:112]: I omitted it deliberately so that we do not have to change it too often especially in unstable packages.

**Pull request merged**